src/modules/endpoint/hocs/withFetcher.js

```diff
@@ -93,13 +83,13 @@ export function withFetcher (endpointName, FailureComponent = Failure, hideSpinn
           }
         }
 
-        if (this.errorVisible()) {
+        if (this.getStatePayload().error) {
           return FailureComponent ? <FailureComponent error={payload.error} /> : null
         }
 
-        const allProps = ({...this.props, [this.endpoint.stateName]: payload.data})
+        const allProps = ({...this.props, [this.props.endpoint.stateName]: payload.data})
         // Strip all internal data
-        delete allProps[this.endpoint.payloadName]
+        delete allProps.endpoint
         delete allProps.getEndpoint
         delete allProps.state
         delete allProps.requestAction
```

Comment:
why don't we just pass only
```js
[this.props.endpoint.stateName]: payload.data
```
as props to the WrappedComponent instead of adding all props to allProps and deleting them again?

Reply:
Because we *want* to pass all props to the WrappedComponent:
If Component1 needs someInfo as a prop and needs the events payload.data then we want to use it like so:
`const FetchingComponent1 = withFetcher('events')(Component1)`
`<FetchingComponent1 someInfo='someInfo:)' />` and expect that `someInfo` gets passed through the `<Fetcher />` into Component1 along with the events data.

Reply:
do we use this?

Reply:
Yeah we use this at some points. It failed when not passing all the props.